src/ansys/mechanical/core/embedding/logger/linux_api.py

```diff
@@ -0,0 +1,134 @@
+"""Internal mechanical logging linux API.
+
+Note - this is needed for version 2023 R2, where the .NET backend is windows-only
+
+Note - some options are not supported with the API, namely the base directory
+and the log filename
+
+"""
+
+import ctypes
+import logging
+import os
+
+from ansys.mechanical.core.embedding import initializer
+from ansys.mechanical.core.embedding.logger import sinks
+
+
+def _get_dll():
+    installdir = os.environ[f"AWP_ROOT{initializer.INITIALIZED_VERSION}"]
+    dll = ctypes.CDLL(os.path.join(installdir, "aisol/dll/linx64/libAns.Common.WBLogger.so"))
+
+    dll.wb_logger_enable_sink.argtypes = [ctypes.c_int32]
+
+    dll.wb_logger_disable_sink.argtypes = [ctypes.c_int32]
+
+    dll.wb_logger_set_sink_filter_level.argtypes = [ctypes.c_int32, ctypes.c_int32]
+
+    dll.wb_logger_flush.argtypes = []
+
+    dll.wb_logger_set_logging_auto_flush.argtypes = [ctypes.c_int8]
+
+    dll.wb_logger_set_logging_directory.argtypes = [ctypes.c_char_p]
+
+    dll.wb_logger_can_log_message.argtypes = [ctypes.c_int32]
+    dll.wb_logger_can_log_message.restype = ctypes.c_int32
+
+    dll.wb_logger_log_message.argtypes = [ctypes.c_int32, ctypes.c_char_p, ctypes.c_char_p]
+    return dll
+
+
+def _get_sink_id(standard_sink_type: int) -> ctypes.c_int32:
+    """Convert standard sink type to sink id."""
+    return {
+        sinks.StandardSinks.STANDARD_LOG_FILE: ctypes.c_int32(1),
+        sinks.StandardSinks.CONSOLE: ctypes.c_int32(2),
+    }[standard_sink_type]
+
+
+def _str_to_utf8_ptr(value: str) -> ctypes.c_char_p:
+    return ctypes.create_string_buffer(value.encode())
+
+
+def _bool_to_single_byte_int(value: bool) -> ctypes.c_int8:
+    if value:
+        return ctypes.c_int8(1)
+    return ctypes.c_int8(0)
+
+
+def _to_wb_logger_severity(level: int) -> ctypes.c_int32:
+    """Convert to internal int."""
```

Comment:
```suggestion
    """Convert to internal integer."""
```
This assumes that "int" meant integer!